plugins/nordic-hid/fu-nordic-hid-cfg-channel.c

```diff
@@ -1034,10 +1034,9 @@ fu_nordic_hid_cfg_channel_setup(FuDevice *device, GError **error)
 	fu_device_add_instance_strsafe(device, "BL", self->bl_name);
 	fu_device_add_instance_strsafe(device, "GEN", self->generation);
 
-	if (self->vid != 0x00 && self->pid != 0x00) {
-		fu_device_add_instance_u16(device, "VEN", self->vid);
-		fu_device_add_instance_u16(device, "DEV", self->pid);
-	}
+	/* peripherals without devinfo really do use hardcoded VID and PID of 0x0000 */
```

Comment:
Can we use the hardcoded values only for device that is connected via a dongle (in such case, the peripheral is a child device of the dongle)? For directly connected devices, reported PID and VID should be proper, because host can get information about VID and PID.

Reply:
isn't what this patch does? i.e. we don't *set* the VID to 0x0 we just use whatever udev told us. It's kinda hard for me to debug this without actual hardware tbh.

Reply:
Can you commit to this branch what works please? i.e. with a device without a devinfo section attached to a dongle.

Reply:
Thanks! :+1:

Reply:
I checked locally that the proposed fix sets VID and PID to 0x0000 also for directly connected devices (which is not intended). I will try to slightly update it.

Reply:
Fix is available here: https://github.com/fwupd/fwupd/pull/5742 (very small change :) )